.circleci/config.yml

```diff
@@ -1,314 +1,232 @@
+    - run_with_coverage:
+        cmd: |
+               cargo install --debug --no-default-features --features="${FEATURES}" --force --path plume-cli ||
+               cargo install --debug --no-default-features --features="${FEATURES}" --force --path plume-cli ||
+               cargo install --debug --no-default-features --features="${FEATURES}" --force --path plume-cli
+    - run:
+        name: run test
+        command: ./script/run_browser_test.sh
+        environment:
+          BROWSER: firefox
+    - upload_coverage:
+        type: integration
+    - cache:
+        cache: <<#parameters.postgres>>postgres<</ parameters.postgres>><<^parameters.postgres>>sqlite<</parameters.postgres>>
+
+  release:
+    parameters:
+      postgres:
+        type: boolean
+    executor:
+      name: default
+      postgres: << parameters.postgres >>
     steps:
-      - run: /bin/true
-  release_postgresql:
-    <<: *defaults
-    <<: *env_postgresql
-    <<: *release
-  release_sqlite:
-    <<: *defaults
-    <<: *env_sqlite
-    <<: *release
+    - restore_env:
+        cache: release-<<#parameters.postgres>>postgres<</ parameters.postgres>><<^parameters.postgres>>sqlite<</parameters.postgres>>
+    - run: cargo web deploy -p plume-front --release
+    - build:
+        package: plume
+        release: true
+    - build:
+        package: plume-cli
+        release: true
+    - cache:
+        cache: release-<<#parameters.postgres>>postgres<</ parameters.postgres>><<^parameters.postgres>>sqlite<</parameters.postgres>>
+    - run: ./script/generate_artifact.sh
+    - unless:
+        condition: << parameters.postgres >>
+        steps:
+        - run: ./script/upload_test_environment.sh
+    - store_artifacts:
+        path: plume.tar.gz
+        destination: plume.tar.gz
+
 
 workflows:
   version: 2
-  build_and_test:
+  build and test:
     jobs:
-      - download_deps
-      - cargo_fmt:
-          requires:
-            - download_deps
-      - build_web:
-          requires:
-            - download_deps
-      - test_cli_postgresql:
-          requires:
-            - download_deps
-      - test_cli_sqlite:
-          requires:
-            - download_deps
-      - test_unit_postgresql:
-          requires:
-            - download_deps
-      - test_unit_sqlite:
-          requires:
-            - download_deps
-      - test_browser_postgresql:
-          requires:
-            - build_web
-      - test_browser_sqlite:
-          requires:
-            - build_web
-      - all_ok:
-          requires:
-            - cargo_fmt
-            - test_cli_postgresql
-            - test_cli_sqlite
-            - test_unit_postgresql
-            - test_unit_sqlite
-            - test_browser_postgresql
-            - test_browser_sqlite
-      - release_postgresql:
-          requires:
-            - all_ok
-      - release_sqlite:
-          requires:
-            - all_ok
+    - cargo fmt
+    - clippy:
+        postgres: false
+    - clippy:
```

Comment:
Could it be possible to give more distinct name to these? I think that because of that we get one job called "clippy" (or whatever) and another one "cliipy-1", but it is not clear which one uses SQlite or PostgreSQL.

Reply:
not without having to duplicate config sadly :/

Reply:
OK, let's forget about it then, it was more bonus.

Reply:
kinda related, I'd prefer to use an enum telling we should use sqlite or postgres, but there is no way to do conditional based on other things than boolean :/
And there is no way to select if we use a docker or not, just to select which one, so I run empty alpine when nothing is needed.
I feel like this new syntax is better than the previous, but it misses some core features to not feel hacky in many places